patches/tModLoader/Terraria/release_extras/start-tModLoaderServer.sh

```diff
@@ -1,25 +1,60 @@
 #!/usr/bin/env bash
-cd "$(dirname "$0")"
+cd "$(dirname "$0")" || exit
 
-launch_args="$@ -server"
-if [[ ! "$launch_args" == *"-config"* ]]; then
+launch_args="-server"
+
+# Parse parameters, passing through everything else to ScriptCaller.sh
+while [[ $# -gt 0 ]]
+do
+	case $1 in
+		-steam)
+			steam_server=true
+			shift
+			;;
+		-nosteam)
+			steam_server=false
+			shift
+			;;
+		*) # If it's not one of the args above, then pass it to ScriptCaller.sh
+			launch_args="$launch_args $1"
+			shift
+			;;
+	esac
+done
+
+# Use serverconfig.txt as config if not already specified
+if ! [[ "$launch_args" == *"-config"* ]]
+then
 	launch_args="$launch_args -config serverconfig.txt"
 fi
 
-read -p "Use Steam Server (y)/(n) " steam
+# Prompt user for lobby type and steam server if not specified in args
+if ! [[ -v steam_server ]]
+then
+	read -p "Use steam server (y/n): " steam_server_response
+	if [[ $steam_server_response == y* ]]
+	then
+		steam_server=true
+	else
+		steam_server=false
+	fi
+fi
 
-if [ $steam = "y" ]; then
+if $steam_server
+then
 	launch_args="$launch_args -steam"
-	
-	clear
-	read -p "Select Lobby Type (f)riends/(p)rivate " lobby
-	
-	if [ $lobby == "f" ]; then 
-		launch_args="$launch_args -lobby friends"
-	else
-		launch_args="$launch_args -lobby private"
+
+	if ! [[ "$launch_args" == *"-lobby"* ]]
+	then
+		read -p "Select lobby type ([f]riends/[p]rivate): " lobby_type_response
+		if [[ $lobby_type_response == f* ]]
+		then
+			launch_args="$launch_args -lobby friends"
+		else
+			launch_args="$launch_args -lobby private"
+		fi
 	fi
 fi
 
 chmod +x ./LaunchUtils/ScriptCaller.sh
-./LaunchUtils/ScriptCaller.sh $launch_args
+./LaunchUtils/ScriptCaller.sh "$launch_args"
```

Comment:
How much has this been tested? have we supplied other args to verify it works fine still?

Reply:
Accidental.